### PR TITLE
fix: updating example.md

### PR DESCRIPTION
**Pull request by tasneemkoushar**
Updated the prerequisite step of the reduce/example.md

This PR is related to the issue #1170

### CODE REVIEW

File: docs/user-guide/user-defined-functions/reduce/examples.md [hunk 1]
```diff
@@ -4,14 +4,45 @@ Please read [reduce](./reduce.md) to get the best out of these examples.
...
 ## Prerequisites
 
-Install the ISB
+#### What is ISB?
+
+`Inter-Step Buffer (ISB)` Service is the service to provide `Inter-Step Buffers`.
+A Pipeline contains multiple vertices that ingest data from sources, process data, and forward processed data to sinks. Vertices are not connected directly but through Inter-Step Buffers. Please refer to the doc [Intern-Step Buffer Service](https://numaflow.numaproj.io/core-concepts/inter-step-buffer-service/) for more information on ISB.
+
+
+#### How to install the ISB
```
Comment (chandankumar4) [inline docs/user-guide/user-defined-functions/reduce/examples.md:13]:
```suggestion
### Inter-Step Buffer (ISB)

#### What is ISB?
An Inter-Step Buffer Service is described by a [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/), which is used to pass data between vertices of a numaflow pipeline.
Please refer to the doc [Intern-Step Buffer Service](../../../core-concepts/inter-step-buffer.md) for more information on ISB.


#### How to install the ISB
```

File: docs/user-guide/user-defined-functions/reduce/examples.md [hunk 2]
```diff
@@ -4,14 +4,45 @@ Please read [reduce](./reduce.md) to get the best out of these examples.
...
+isbsvc-default-js-0                          3/3     Running     0          19s
+isbsvc-default-js-1                          3/3     Running     0          19s
+isbsvc-default-js-2                          3/3     Running     0          19s
+simple-pipeline-daemon-78b798fb98-qf4t4      1/1     Running     0          10s
+simple-pipeline-out-0-xc0pf                  1/1     Running     0          10s
+simple-pipeline-cat-0-kqrhy                  2/2     Running     0          10s
+simple-pipeline-in-0-rhpjm                   1/1     Running     0          11s
+```
+
+**_NOTE:_**  The Source used in the examples is an HTTP source producing messages with values 5 and 10 with event time
```
Comment (chandankumar4) [inline docs/user-guide/user-defined-functions/reduce/examples.md:44]:
```suggestion
---
**NOTE**

The Source used in the examples is an HTTP source producing messages with values 5 and 10 with event time
```

Comment (chandankumar4) [reply docs/user-guide/user-defined-functions/reduce/examples.md:44]: Also add `---` in line no: 52

**Review by chandankumar4 — CHANGES_REQUESTED** [review]

File: docs/user-guide/user-defined-functions/reduce/examples.md [hunk 3]
```diff
@@ -4,14 +4,45 @@ Please read [reduce](./reduce.md) to get the best out of these examples.
...
+
+NAME                                         READY   STATUS      RESTARTS   AGE
+isbsvc-default-js-0                          3/3     Running     0          19s
+isbsvc-default-js-1                          3/3     Running     0          19s
+isbsvc-default-js-2                          3/3     Running     0          19s
+simple-pipeline-daemon-78b798fb98-qf4t4      1/1     Running     0          10s
+simple-pipeline-out-0-xc0pf                  1/1     Running     0          10s
+simple-pipeline-cat-0-kqrhy                  2/2     Running     0          10s
+simple-pipeline-in-0-rhpjm                   1/1     Running     0          11s
+```
```
[the remark this answers is not in the document]
Comment (tasneemkoushar) [reply docs/user-guide/user-defined-functions/reduce/examples.md]: Have removed this in latest commit

File: docs/user-guide/user-defined-functions/reduce/examples.md [hunk 4]
```diff
@@ -4,14 +4,41 @@ Please read [reduce](./reduce.md) to get the best out of these examples.
...
+isbsvc-default-js-2                          3/3     Running     0          19s
+```
+
+---
+**NOTE**
+
+The Source used in the examples is an HTTP source producing messages with values 5 and 10 with event time
+starting from 60000. Please refer to the doc [http source](../../sources/http.md) on how to use an HTTP
 source.
 An example will be as follows,
```
Comment (chandankumar4) [inline docs/user-guide/user-defined-functions/reduce/examples.md:44]: can you add `---` in line no: 49 (similar to line no: 37)

Comment (tasneemkoushar) [reply docs/user-guide/user-defined-functions/reduce/examples.md:44]: Added in the latest commit

**Review by chandankumar4 — APPROVED** [review]

File: docs/user-guide/user-defined-functions/reduce/examples.md [hunk 5]
```diff
@@ -4,21 +4,49 @@ Please read [reduce](./reduce.md) to get the best out of these examples.
 
 ## Prerequisites
 
-Install the ISB
+### Inter-Step Buffer (ISB)
```
Comment (whynowy) [inline docs/user-guide/user-defined-functions/reduce/examples.md:7]: Inter-Step Buffer Service (ISB Service)

File: docs/user-guide/user-defined-functions/reduce/examples.md [hunk 6]
```diff
@@ -4,21 +4,49 @@ Please read [reduce](./reduce.md) to get the best out of these examples.
...
+
+#### What is ISB?
```
Comment (whynowy) [inline docs/user-guide/user-defined-functions/reduce/examples.md:9]: ISB Service

File: docs/user-guide/user-defined-functions/reduce/examples.md [hunk 7]
```diff
@@ -4,21 +4,49 @@ Please read [reduce](./reduce.md) to get the best out of these examples.
...
+An Inter-Step Buffer Service is described by a [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/), which is used to pass data between vertices of a numaflow pipeline.
+Please refer to the doc [Intern-Step Buffer Service](../../../core-concepts/inter-step-buffer.md) for more information on ISB.
+
+
+#### How to install the ISB
```
Comment (whynowy) [inline docs/user-guide/user-defined-functions/reduce/examples.md:14]: ISB Service

File: docs/user-guide/user-defined-functions/reduce/examples.md (hunk 5 above)
Comment (tasneemkoushar) [reply docs/user-guide/user-defined-functions/reduce/examples.md]: Changed in the latest commit.

File: docs/user-guide/user-defined-functions/reduce/examples.md (hunk 6 above)
Comment (tasneemkoushar) [reply docs/user-guide/user-defined-functions/reduce/examples.md]: changed in my latest commit

File: docs/user-guide/user-defined-functions/reduce/examples.md (hunk 7 above)
Comment (tasneemkoushar) [reply docs/user-guide/user-defined-functions/reduce/examples.md]: Changed in my latest commit

**Review by whynowy — APPROVED** [review]

**Pull request merged**